invenio_vocabularies/datastreams/readers.py

```diff
@@ -116,6 +117,32 @@ def read(self, item=None, *args, **kwargs):
         yield from self._iter(url=url, *args, **kwargs)
 
 
+class RORHTTPReader(BaseReader):
+    """ROR HTTP Reader returning an in-memory binary stream of the latest ROR data dump ZIP file."""
+
+    def _iter(self, fp, *args, **kwargs):
+        raise NotImplementedError(
+            "RORHTTPReader downloads one file and therefore does not iterate through items"
+        )
+
+    def read(self, item=None, *args, **kwargs):
+        """Reads the latest ROR data dump ZIP file from Zenodo and yields an in-memory binary stream of it."""
+        if item:
+            raise NotImplementedError(
+                "RORHTTPReader does not support being chained after another reader"
+            )
+
+        api_url = "https://zenodo.org/api/communities/ror-data/records?q=&sort=newest"
```

Comment:
Even safer maybe would be to use the Concept DOI of the record, though I don't think we're aiming at proper data citation practices here :)

In any case, using the DOI would be an issue since it resolves to the UI/HTML version of the latest record, so there's no easy way on how to access the machine-friendly format. IIRC Signposting might solve that...

Edit: TIL we stealth-deployed signposting on Zenodo thanks to @fenekku 🤗

You can run the following to get links to all files of the latest record using:

```bash
$ curl -I -L  "https://doi.org/10.5281/zenodo.6347574"
HTTP/2 302
location: https://zenodo.org/doi/10.5281/zenodo.6347574
...

HTTP/1.1 302 FOUND
location: https://zenodo.org/records/11186879
...

HTTP/1.1 200 OK
link: <https://zenodo.org/api/records/11186879> ; rel="linkset" ; type="application/linkset+json"
...

# Using the `Link:` header from the last response with "Accept: application/linkset+json"
curl "https://zenodo.org/api/records/11186879" -H "Accept: application/linkset+json"

{
  "linkset": [
    {
      "anchor": "https://zenodo.org/records/11186879",
      ...
      "item": [
        {
          "href": "https://zenodo.org/records/11186879/files/v1.46.1-2024-05-13-ror-data.zip",
          "type": "application/zip"
        }
      ],
      ...
    },
    ...
  ]
}
```

Reply:
I should have probably added it as a comment in the code: I followed the [Download ROR data dumps programmatically with the Zenodo API](https://ror.readme.io/docs/data-dump#download-ror-data-dumps-programmatically-with-the-zenodo-api) documentation from ROR, which does recommend to find the latest record in the community.

Happy to use the signpost approach suggested by @slint if you think it's better.

Reply:
Oh cool! I'd do it all in one step then `curl "https://zenodo.org/api/records/6347574" -L -H "Accept: application/linkset+json"` and then go down to item.

Someone should probably reach out to ROR and get them to change their instructions. The linkset call will make the download instructions much easier and more robust (just one call with a standard structure). I could do the outreach.....but I'm not really a Zenodo representative.

Reply:
Done, I am now calling the Concept DOI with `Accept: application/linkset+json` and then trying to find the one ZIP file.
Please have a look at the updated method body.